zookeeper-server/src/main/java/org/apache/zookeeper/server/DataTree.java

```diff
@@ -202,8 +212,8 @@
         return new HashSet<String>(containers);
     }
 
-    public Set<String> getTtls() {
-        return new HashSet<String>(ttls);
+    public Set<TTLNode> getTtls() {
+        return new LinkedHashSet<>(ttls);
```

Comment:
Is the desire here not to allocate a potentially large internally array when there are many TTL nodes? I think we shouldn't even bother with the defensive copy here. Returning `Collections.unmodifiableSet(ttls)` is fine and, given that this is only an internally used object, just the internal ttls would be OK with an appropriate comment (and maybe renaming the method).